### ISSUE TITLE
Distributed IO - local data invalid

### ISSUE THREAD

**Issue by ndryden** [LLNL/lbann#190]
Continuing my streak of breaking things with the Pilot 1 data...

Running with:
`/usr/workspace/wsa/dryden1/lbann2/./build/gnu.catalyst.llnl.gov/install/bin/lbann --model=pilot1b3_snn_autoencoder.prototext --reader=model_zoo/data_readers/data_reader_candle_pilot1.prototext --optimizer=model_zoo/optimizers/opt_adam.prototext`

with everything default except using 32 ranks, normal imcomm, 16 procs per model. Note the only difference between this and #182 is the `validation_percent` for the training data reader.

I get this error:
```
--------------------------------------------------------------------------------
[0] Epoch : stats formated [tr/v/te] iter/epoch = [645/144/160]
            global MB = [2048/1024/1024] global last MB = [ 806/ 203/ 113]
             local MB = [1024/1024/1024]  local last MB = [ 403/ 203/ 113]
--------------------------------------------------------------------------------

**************************************************************************
 This lbann_exception is about to be thrown:/usr/workspace/wsa/dryden1/lbann2/src/io/data_buffers/distributed_io_buffer.cpp 71 :: lbann_distributed_io_buffer: No valid data for this step -- local data was invalid

 Am now attemptting to print the stack trace ...
**************************************************************************
rank: 20 :: lbann::stack_trace::print_stack_trace()
rank: 20 :: lbann::stack_trace::print_lbann_exception_stack_trace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
rank: 20 :: lbann::distributed_io_buffer::distribute_from_local_matrix(El::AbstractDistMatrix<float>&, lbann::generic_data_reader*, execution_mode)
rank: 20 :: lbann::generic_input_layer::fp_compute()
rank: 20 :: lbann::Layer::forward_prop()
rank: 20 :: lbann::model::forward_prop(execution_mode)
rank: 20 :: lbann::model::train_mini_batch()
rank: 20 :: lbann::model::train(int)
rank: 20 :: dli_sname == NULL for: 0x404055 backtrace message was:      /usr/workspace/wsa/dryden1/lbann2/./build/gnu.catalyst.llnl.gov/install/bin/lbann() [0x404055]
rank: 20 :: demangling failed for: __libc_start_main
rank: 20 :: dli_sname == NULL for: 0x40474e backtrace message was:      /usr/workspace/wsa/dryden1/lbann2/./build/gnu.catalyst.llnl.gov/install/bin/lbann() [0x40474e]
```

Another curious thing, despite throwing this exception, execution _was not terminated_. Maybe something to do with the stack tracing? @davidHysom can you check that?

**Comment by bvanessen** [comment]
Fixed how the distributed mini-batch calculates the offsets in the multi-model case.  Merged in at 48e66cff0a04970dc85230090ab88b924fe0d54b

**Comment by bvanessen** [comment]
Fixed in f1a19d269fbb9d8eb8c72285e06286f11c1449bb
